backend/src/openarchiefbeheer/destruction/models.py

```diff
@@ -154,6 +155,17 @@ def assign_next(self) -> None:
             self.set_status(status)
             return
 
+        # Check if archivist has approved the list.
+        try:
+            archivist = self.assignees.get(role=ListRole.archivist)
+            if archivist and self.assignee == archivist.user:
+                self.get_author().assign()
+                self.set_status(ListStatus.ready_to_delete)
+                return
```

Comment:
This should be outside the try/except.

But thinking about it more: this function is called only when a review is created. Therefore the assignee must be a reviewer or an archivist. Maybe we can start with:
```python
current_assignee = self.assignees.get(user=self.assignee)
```
From `current_assignee.role` we know if we are dealing with reviewer or archivist. I think we can use this to make it easier to follow in which flow we are :thinking:

Reply:
🤔